enigma-swing/src/main/java/cuchaz/enigma/gui/GuiController.java

```diff
@@ -119,6 +133,51 @@ public void closeJar() {
 		this.gui.onCloseJar();
 	}
 
+	public void close() {
+		autoSaveExecutor.shutdown();
+	}
+
+	private synchronized void autoSave() {
+		if (!UiConfig.autoSave() || project == null) {
+			return;
+		}
+
+		final EntryRemapper mapper = project.getMapper();
+		final Path autoSaveDir = ConfigPaths.getConfigPathRoot().resolve("enigma").resolve("autosave");
+
+		try {
+			final Path autoSavePath = autoSaveDir.resolve(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss").format(LocalDateTime.now()) + ".mappings");
```

Comment:
Maybe add name of current project (jar file name, I guess?) to this to make finding which project each file belongs to easier?